golang/cosmos/app/app.go

```diff
@@ -845,6 +845,10 @@ func unreleasedUpgradeHandler(app *GaiaApp, targetUpgrade string) func(sdk.Conte
 		// Each CoreProposalStep runs sequentially, and can be constructed from
 		// one or more modules executing in parallel within the step.
 		CoreProposalSteps := []vm.CoreProposalStep{
+			// First, upgrade wallet factory
+			vm.CoreProposalStepForModules("@agoric/builders/scripts/smart-wallet/build-wallet-factory2-upgrade.js"),
+			// Then, upgrade Zoe and ZCF
+			vm.CoreProposalStepForModules("@agoric/builders/scripts/vats/replace-zoe.js"),
```

Comment:
I'm only vaguely familiar with this `CoreProposalSteps` go stuff. I presume that the mechanism itself is sufficiently tested and the tests in`post.test.js` suffice to show that this usage is wired up right.

Reply:
This is the mechanism now used in the release branch, so this is just replicating that here on the master branch